Replaced the new option in Recovering a Repository topic (#1244)
theforeman/foreman-documentation: guides/doc-Content_Management_Guide/topics/Importing_Content.adoc

Synchronizes all RPMs regardless of detected changes.
Use this option if specific RPMs could not be downloaded to the local repository even though they exist in the upstream repository.

> //Validate Content Sync::

Reviewer: old content should be removed rather than commented out.

. In the {Project} web UI, navigate to *Content* > *Products*.
. Select the product containing the corrupted repository.
. Select the name of a repository you want to synchronize.
. To verify the checksum, click *Verify Content Checksum* from the *Action* menu.

> . To perform optimised sync and complete sync, select *Advanced Sync*, perform the following:

Reviewer: We don't use subsections and I don't see the value of a subsection here as this is as clear when written in a linear procedure. Are you performing one or two syncs? If you are selecting one of two, you need to use or rather than and. 
```suggestion
. To perform optimised sync or complete sync, select *Advanced Sync*.
```

> .. Select *Advanced Sync* from the *Select Action* menu.

Reviewer: seems like a duplicate of the above line?

Author: Yes, it is the same as above for which you have a suggestion.

Reviewer: Then you don't need to say it twice.

> .. Select the desired option.

Reviewer: Do not use sub-steps. There's no reason as the text reads as well and isn't complicated. 
We avoid sub steps if possible. 

IBM Style says never ever use desire in technical writing. 

desired adj
Do not use. Use “required” or “the <item> that you want.”

Is this the point where the user selects from optimised or complete sync? Say that.

Author: Hi @melcorr,

Before committing these changes, I had a session with Shweta. Although she wasn't confident about sub-sections, she felt that we are left with no other options and hence, we opted for this. Writing style and word selection are kept the same except for the changes that we took (the last 2 points).

I will replace the word 'desire' with 'required' as I can see the same on IBM Style Guide Quick Reference.

I will also take the changes on the above suggestion where the 'and' can be replaced with 'or'.

At last, I have not taken any decision on the style guide or sentence structure in this commit as it had minor modifications & this can be the probable reason why I am assigned this type of ticket at the current level.

I can understand from your suggestions that there can be a better writing style and sentence structure. But, the advanced sync and verify content checksum are two different actions that are included in the same drop-down menu which can be the probable reason for confusion, as per my understanding.

Please guide me further to make this commit perfect.

Reviewer: > Hi @melcorr,
> 
> Before committing these changes, I had a session with Shweta. Although she wasn't confident about sub-sections, she felt that we are left with no other options and hence, we opted for this. Writing style and word selection are kept the same except for the changes that we took (the last 2 points).
I don't understand the "left with no other options" comment because I cannot see the complication that this PR is introducing. 
> 
> I will replace the word 'desire' with 'required' as I can see the same on IBM Style Guide Quick Reference.

If you are selecting between two options, this or that, you could jjust say that. "required" is ok but less descriptive.
My main issue is that desired/required aren't descriptive enough. Do you know what that menu reveals at this point?  
> 
> I will also take the changes on the above suggestion where the 'and' can be replaced with 'or'.
> 
> At last, I have not taken any decision on the style guide or sentence structure in this commit as it had minor modifications & this can be the probable reason why I am assigned this type of ticket at the current level.

That's fair. Everything is a learning opportunity all the same. 
> 
> I can understand from your suggestions that there can be a better writing style and sentence structure. But, the advanced sync and verify content checksum are two different actions that are included in the same drop-down menu which can be the probable reason for confusion, as per my understanding.
> 
> Please guide me further to make this commit perfect.

We've avoided sub-procedures everywhere so far. If all the steps are happening in a sequence, then they can remain in that same sequence. If it is an optional step, which means it mightn't happen at all, it would need to be marked as optional. If the user must choose something, tell the user what the options are. 

I hope this is helping a bit, Malhar.  I understand that you're learning and eager to learn.

Author: > If you are selecting between two options, this or that, you could jjust say that. "required" is ok but less descriptive.
> My main issue is that desired/required aren't descriptive enough. Do you know what that menu reveals at this point?

I think the description of all 3 options are there just before the procedure steps. So my question is - the description will still require in the procedure step as well?

> We've avoided sub-procedures everywhere so far. If all the steps are happening in a sequence, then they can remain in that same sequence. If it is an optional step, which means it mightn't happen at all, it would need to be marked as optional. If the user must choose something, tell the user what the options are.
> 
> I hope this is helping a bit, Malhar. I understand that you're learning and eager to learn.

Verify Content Checksum can be the optional step as per my understanding. We can mention it as " (optional) " in procedure steps as per your suggestion. However, I will clear these queries once again with peers.

So, at the end:

1) I will replace require with desire.
2) I will re-verify the options which are mandatory and optional..
3) Sub-steps will be removed and will cover those steps in a single point, even if it requires one more point to be added.

Please let me know if I am missing something.

. In the {Project} web UI, navigate to *Content* > *Products*.
. Select the product containing the corrupted repository.
. Select the name of a repository you want to synchronize.

> . From the *Select Action* menu, select *Advanced Sync*.

Reviewer 2: I think this is still relevant, as optimized sync and complete sync are still accessible after clicking on Advanced sync. The option here refers to what the user might want to select, either optimized or complete. The section after this elaborates on the meaning of the different options. It probably might be better to add another section for "Verify Content Checksum" outside of the advanced sync section.

> . To verify the checksum, click *Verify Content Checksum* from the *Action* menu.

Reviewer 2: . To verify the checksum, click *Verify Content Checksum* from the *Select Action* menu.

Reviewer: Orient the user first as per IBM style, so @mjivraja is slightly more correct here. Depending on the style guide, this rule changes, but we follow IBM here.

> . To perform optimised sync and complete sync, select *Advanced Sync* from the *Select Action* menu. And then, select the option and click *Sync*.

Reviewer: We use American English in the docs so ized rather than ised. 
We don't start paragraphs with And. 
What option? we should tell them what to select.
